lib/workspace.rb

```diff
@@ -0,0 +1,57 @@
+require "httparty"
+require_relative "../lib/user"
+require_relative "../lib/channel"
+require_relative "../lib/recipient"
+require "json"
+
+class Workspace
+  attr_reader :users, :channels, :selected
+
+  def initialize(selected:)
+    @users = User.list
+    @channels = Channel.list
+    @selected = selected # either user or channel info
+  end
+
+  def select_user
+    user_selected = users.detect do |user|
+      user.slack_id == selected || user.name == selected
+    end
+    return user_selected
+  end
+
+  def select_channel
+    channel_selected = channels.detect do |channel|
+      channel.slack_id == selected || channel.name == selected
+    end
```

Comment:
This code is very similar to the code for `select_user` above. Could you DRY this up somehow?